pyrsa/data/dataset.py

```diff
@@ -299,7 +299,37 @@ def sort_by(self, by):
         order = np.argsort(desc)
         self.measurements = self.measurements[order]
         self.obs_descriptors = subset_descriptor(self.obs_descriptors, order)
+        
+    def get_measurements(self):
+        "Getter function for measurements"
+        return self.measurements.copy()
+    
+    def get_measurements_tensor(self, by):
+        """ Returns a tensor version of the measurements array, split by an 
+        observation descriptor
 
+        Args:
+            by(String):
+                the descriptor by which the splitting is made
+
+        Returns:
+            measurements_tensor (ndarray):
+                3d array with the selected obs_descriptor constituting the
+                third dimension.
+        """
+        assert len(self.obs_descriptors.keys()) > 1, \
+            "dataset has too few obs_descriptors to build a tensor from it"
```

Comment:
Why couldn't I want to build a tensor from a dataset with only one descriptor?

Reply:
Because (unless you have multiple measurements with identical descriptors,) the dimensions of the tensor would be 1  x n_channels x n_obs_by.

Reply:
First: I think this line checks whether there multiple descriptors not whether there are multiple values. So, there are cases where I would end up with a perfectly fine array and cannot get it because this assert fails.
Second: Even if I end up with a 1 x n_channels x n_obs array that might well be what I want to do in some cases for example if the number of runs differs between subjects and some subjects have only one run.